src/System.Security.Cryptography.Primitives/src/Resources/Strings.resx

```diff
@@ -105,6 +105,9 @@
   <data name="Cryptography_InvalidPaddingMode" xml:space="preserve">
     <value>Specified padding mode is not valid for this algorithm.</value>
   </data>
+  <data name="Cryptography_InvalidHashAlgorithmOid" xml:space="preserve">
+    <value>Specified OID is not a valid hash algorithm.</value>
```

Comment:
Should the error message include the invalid value?

Reply:
I like that idea. None of the other exceptions in the resource file do that though, I wasn't sure if that was intentional.

Reply:
None in this one do, but our newer exceptions tend to be a bit nicer, like from CMS:

```xml
  <data name="Cryptography_Cms_UnknownAlgorithm" xml:space="preserve">
    <value>Unknown algorithm '{0}'.</value>
  </data>
```

So I'd go with something like

```suggestion
    <value>The specified OID ({0}) does not represent a known hash algorithm.</value>
```